### PR TITLE
fix: [CDS-82953]: Add force delete option for aws, azure, gcp, k8s, rancher, oci and helm connectors

### PR DESCRIPTION
**Pull request by abosii-harness**
## Describe your changes
This changes provide the force_delete option for next connectors:
- [x] GCP
- [x] AWS
- [x] Azure
- [x] Kubernetes
- [x] Rancher
- [x] Oci Helm
- [x] Http Helm

## Comment Triggers

<details>
  <summary>PR Check triggers</summary>
  
- Build: `trigger build`
- Sub Category Field Check: `trigger subcategoryfieldcheck`
- gitleaks: `trigger gitleaks`

### CODE REVIEW

**Comment by abosii-harness** [comment]
Doc ticket: https://harness.atlassian.net/browse/DOC-3888

File: internal/service/platform/connector/azure_cloud_provider.go [hunk 1]
```diff
@@ -166,6 +166,12 @@ func ResourceConnectorAzureCloudProvider() *schema.Resource {
 				Type:        schema.TypeBool,
 				Optional:    true,
 			},
+			"force_delete": {
+				Description: "Enable this flag for force deletion of connector",
+				Type:        schema.TypeString,
```
Comment (acasian-harness) [inline internal/service/platform/connector/azure_cloud_provider.go:171]: shouldn't it by boolean?

**Review by acasian-harness — APPROVED** [review]

File: internal/service/platform/connector/azure_cloud_provider.go (hunk 1 above)
Comment (abosii-harness) [reply internal/service/platform/connector/azure_cloud_provider.go:171]: copy paste, not sure why in other places this is string, checking

Comment (abosii-harness) [reply internal/service/platform/connector/azure_cloud_provider.go]: there was no reason to put it as string before, update to bool

**Review by abhinavsingh1196 — APPROVED** [review]

**Review by GoelPratyush99 — APPROVED** [review]

**Review by TarunUba1 — APPROVED** [review]

**Review by rathodmeetsatish — CHANGES_REQUESTED** [review]
include connector tests in terraform cd pipeline

**Review by rathodmeetsatish — APPROVED** [review]

**Pull request merged**